lib/controllers/users.js

```diff
@@ -0,0 +1,178 @@
+'use strict';
+const boom = require('boom');
+const joi = require('joi');
+const _ = require('lodash');
+const moment = require('moment');
+
+const internals = {};
+
+/**
+ * Test whether a user with at least one of the properties exists
+ * @param  {object} authUtils an instance of the authentication utilities
+ * @param  {object} userModel bookshelf model to fetch user data
+ * @param  {object} userData  object containing properties to match the user on
+ * @return {object}          undefined if no matches found,
+ *                                 otherwise a bookshelf object
+ */
+internals.getUser = (authUtils, UserModel, userData) => {
+    const user = new UserModel();
+    const formatQuery = user._utils.formatQuery;
+
+    //clone to avoid mutation
+    let username = userData.username;
+    let qbObj = {orWhere: formatQuery(userData)};
+    if (username) {
+        qbObj.orWhere.username = authUtils.encryptUsername(username);
+    }
+
+    return user
+        .query(qbObj)
+        .fetch();
+};
+
+internals.defaultValues = {
+    siteId: '-1',
+    activeFlag: 'Y',
+    passwordExpDate: moment().add(1, 'years').format('MM/DD/YYYY'),
+    acctExpDate: moment().add(1, 'years').format('MM/DD/YYYY')
+};
+
+/**
+ * Call the toJSON method on the object
+ * @param  {Object} obj object to call the toJSON method on (bookshelf)
+ * @return {string}     JSON
+ */
+internals.callToJson = (obj) => {
```

Comment:
We may be able to modify hapi's `reply` functionality to do this automatically. Alternatively, we could place this as a utility on `server` so that it would not need to be defined everywhere.